Add use case documentation (#6)
markphelps/flipt.io: content/docs/use_cases.md

# Use Cases

Flipt provides a solution for **feature flagging**. The next few sections provide examples of how Flipt can be used.

## A/B Testing

> A/B testing can be accomplished by setting up a percentage rollout in Flipt. First you will need to create a segment. This is detailed in [Getting Started]({{< relref "/docs/getting_started" >}}). You can simply create a segment with no constraints, which will match all users.

Reviewer: ```suggestion
A/B testing can be accomplished by setting up a percentage rollout rule in Flipt. See the [Getting Started]({{< relref "/docs/getting_started" >}}) section for how accomplish this.
```

> Next you will want to create a flag with some variants by going to **Flags -> New Flag**. For this example I have created two variants: group-a and group-b.

Reviewer: ```suggestion
In this example rule, we have two variants that we are targeting, `group-a` and `group-b`, each with a 50% distribution.
```

> After you have both your segment and flag created, head on over to the **Targeting** tab in the flag that you just made. Then select **New Rule**. From here you can match on your newly created segment and choose **A Percentage Rollout** from the dropdown.

Reviewer: ```suggestion
```

> Next, choose your percentages like so:

Reviewer: ```suggestion
```

{{< figure src="/img/use_cases/flag_targeting_rollout.png" alt="Flag Targeting" >}}

> Now that your A/B testing flag is all set up you can begin to make requests! Using your favorite Flipt gRPC client library, you can call the `Evaluate` method with the request parameters.

Reviewer: ```suggestion
An example request to evaluate this rule within your application would look like this:
```

*Language: Go*
```go
res, err := client.Evaluate(context.Background(), &flipt.EvaluationRequest{
		FlagKey:  "new-landing-page",
		EntityId: "test@email.com",
	})
```

{{< hint info >}}
Notice that we are using an email address for the `EntityId`. This does not need to be an email address, it just needs to be a unique identifier for your users.
{{< /hint >}}

> One of the fields in the reulting struct is `Value`. You can check this to determine what landing page to respond with:

Reviewer: ```suggestion
The response will contain which variant we should show for this user, either `group-a` or `group-b`:
```
